examples/flutter_gallery/lib/demo/material/text_form_field_demo.dart

```diff
@@ -116,103 +171,113 @@ class TextFormFieldDemoState extends State<TextFormFieldDemo> {
-                inputFormatters: <TextInputFormatter> [
-                  WhitelistingTextInputFormatter.digitsOnly,
-                  // Fit the validating format.
-                  _phoneNumberFormatter,
-                ],
-              ),
-              new TextFormField(
-                decoration: const InputDecoration(
-                  icon: const Icon(Icons.email),
-                  hintText: 'Your email address',
-                  labelText: 'E-mail',
+                const SizedBox(height: 24.0),
+                new TextFormField(
+                  decoration: const InputDecoration(
+                    border: const UnderlineInputBorder(),
+                    filled: true,
+                    icon: const Icon(Icons.email),
+                    hintText: 'Your email address',
+                    labelText: 'E-mail',
+                  ),
+                  keyboardType: TextInputType.emailAddress,
+                  onSaved: (String value) { person.email = value; },
                 ),
-                keyboardType: TextInputType.emailAddress,
-                onSaved: (String value) { person.email = value; },
-              ),
-              new TextFormField(
-                decoration: const InputDecoration(
-                  hintText: 'Tell us about yourself',
-                  helperText: 'Keep it short, this is just a demo',
-                  labelText: 'Life story',
+                const SizedBox(height: 24.0),
+                new TextFormField(
+                  decoration: const InputDecoration(
+                    border: const OutlineInputBorder(),
+                    hintText: 'Tell us about yourself',
+                    helperText: 'Keep it short, this is just a demo.',
+                    labelText: 'Life story',
+                  ),
+                  maxLines: 3,
+                ),
+                const SizedBox(height: 24.0),
+                new TextFormField(
+                  keyboardType: TextInputType.number,
+                  decoration: const InputDecoration(
+                    border: const OutlineInputBorder(),
+                    labelText: 'Salary',
+                    prefixText: '\$',
+                    suffixText: 'USD',
+                    suffixStyle: const TextStyle(color: Colors.green)
+                  ),
+                  maxLines: 1,
                 ),
-                maxLines: 3,
-              ),
-              new TextFormField(
-                keyboardType: TextInputType.number,
-                decoration: const InputDecoration(
-                  labelText: 'Salary',
-                  prefixText: '\$',
-                  suffixText: 'USD',
-                  suffixStyle: const TextStyle(color: Colors.green)
+                const SizedBox(height: 24.0),
+                new PasswordField(
+                  fieldKey: _passwordFieldKey,
+                  helperText: 'No more than 8 characters.',
+                  labelText: 'Password *',
```

Comment:
Don't you want to add "* required" to the helperText?  Either that, or take away the asterisk, I think.